File: api/model/menu.go
package model

import "gorm.io/gorm"

type Menu struct {
	gorm.Model
	GroupID     uint
	Level       string
	Aim         string
	Description string
}

type Practice struct {
	gorm.Model
	MenuID      int
	Menu        Menu
	Category    string
	System      string
	Distance    string
	Times       string
	Set         string
	OnTime      string
	Description string
}

type User struct {
	gorm.Model
	Name    string
	Sex     string
	Email   string
	GroupID uint
}

type Group struct {
	gorm.Model
	Name  string
	Users []User
	Menus []Menu
}

File: entities/Menu.ts
export type Group = {
  id: number
  menus: Menu[]
  name: string
}

export type Menu = {
  id: number
  level: string
  aim: string
  description: string
  createdAt: string
  group: Partial<Group>
  practices: Practice
  updatedAt: string
}

export type Practice = {
  category: String
  createdAt: string
  description: string
  distance: string
  id: number
  menu: Menu
  onTime: string
  set: number
  system: string
  times: number
  updatedAt: string
}
